server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java

```diff
@@ -129,9 +128,31 @@ public boolean writeNextObject(OneRow onerow) throws IOException {
             // match same buffer size to read data from the input stream. The
             // buffer size can be configured externally.
             int bufferSize = configuration.getInt("io.file.buffer.size", DEFAULT_BUFFER_SIZE);
-            long count = IOUtils.copyLarge((InputStream) onerow.getData(), dos, new byte[bufferSize]);
-            LOG.debug("Wrote {} bytes to outputStream using a buffer of size {}", count, bufferSize);
-            return count > 0;
+
+            InputStream inputStream = (InputStream) onerow.getData();
+            final byte[] buffer = new byte[bufferSize];
+
+            // The logic below is copied from IOUtils.copyLarge to add logging
+            long totalByteCount = 0;
+            int n;
+            long iterationCount = 0;
+            long iterationByteCount = 0;
+
+            while (-1 != (n = inputStream.read(buffer))) {
+                dos.write(buffer, 0, n);
+                totalByteCount += n;
+                iterationByteCount += n;
+                if (LOG.isDebugEnabled() && (iterationCount % 100 == 0)) {
+                    // Log this message after every 100th Iteration
+                    LOG.debug("wrote {} bytes, total so far {} (buffer size {})", iterationByteCount, totalByteCount, bufferSize);
+                    iterationByteCount = 0;
+                }
+
+                iterationCount++;
+            }
+
+            LOG.debug("Wrote {} bytes to outputStream using a buffer of size {}", iterationCount, bufferSize);
+            return iterationCount > 0;
```

Comment:
while this is correct, should we keep it in the spirit of the original code that checked the number of bytes written (vs number of iterations ?)
```suggestion
            return totalByteCount > 0;
```